### DISCUSSION TITLE
Compressor specs in dynamics mode

### DISCUSSION THREAD

**Discussion by svidille** [DanWBR/dwsim#591] [General]
Dear Dan and dear all,

When there are two pressure specs fixed, should the flow be calculated? In the attached example, no matter if pressure or flow are fixed, pressure and flow in the outlet cannot be specified, only calculated.

The example is for compressor but maybe it is good to know the philosophy behind the specs in dynamics mode in DWSIM.

[5.Compressor-PF - copia.zip](https://github.com/DanWBR/dwsim/files/14024927/5.Compressor-PF.-.copia.zip)

Thanks,
Simone

**Comment by DanWBR** [comment]
@svidille you can check the source code by yourself. In DWSIM.UnitOperations project you'll find the files for each unit operation, and then the RunDynamicModel() routine inside them.

**Comment by svidille** [comment]
Hi, Dan, thanks so much for your answer. 

I am on that. I suppose the root of the errors and questions in Dynamics is the misunderstanding of the concept of P/F boundaries in DWSIM, which is basically different from other simulators. 

I will try to behave myself and only send one issue a day, as I totally understand your point of being alone on answering all our questions.

Best wishes,
Simone

**Comment by DanWBR** [comment]
@svidille I'm not an expert in dynamics myself, so I can't compare my implementation against other simulators also because I don't have a license for any of them. I advise you to slow down a little bit and read the user's guide, that's the best I could do so far.

The P/F boundaries in DWSIM are basically used for valves. Please check this part of the source code: https://github.com/DanWBR/dwsim/blob/windows/DWSIM.UnitOperations/UnitOperations/Valve.vb#L302-L440

**Comment by DanWBR** [comment]
@svidille as an additional information, try not to compare DWSIM Dynamics with other dynamic simulators. Many unit operations are not supported and some are not complete, like the compressor and expander models which are exactly the same as their steady-state counterparts.
